test/test/lock/RaceToLock.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package test.lock;
+
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.concurrent.locks.LockSupport;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+public class RaceToLock {
+
+    private final Lock sharedLock = new ReentrantLock(true);
+    private final Object sharedObj = new Object();
+    private long sharedWaitTime = 0;
+
+    private final Random random = new Random();
+    private final Lock lock1 = new ReentrantLock(true);
+    private final Lock lock2 = new ReentrantLock(true);
+    private volatile long randomWaitTime = 0;
+
+    private volatile boolean exitRequested = false;
```

Comment:
nit: it does not make difference here, but generally initializing volatile fields with default values is a bad practice.